toolkit/tools/pkg/imagecustomizerlib/shrinkfilesystems.go

```diff
@@ -0,0 +1,151 @@
+package imagecustomizerlib
+
+import (
+	"fmt"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/imagegen/diskutils"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/logger"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/shell"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+func shrinkFilesystems(imageLoopDevice string, outputImageFile string) error {
+	logger.Log.Infof("Shrinking filesystems")
+
+	// Get partition info
+	diskPartitions, err := diskutils.GetDiskPartitions(imageLoopDevice)
+	if err != nil {
+		return err
+	}
+
+	// Get the start sectors of all partitions
+	matchStarts, err := getStartSectors(imageLoopDevice, len(diskPartitions)-1)
```

Comment:
curious, why the `-1`?

Reply:
the first thing in the diskPartitions object refers to the image itself... not a partition. so the number of partitions is len(diskPartitions)-1

Reply:
got it - thx Amrita... maybe we can add a comment...

Reply:
added